File: hw/Pérez/U3/HW29TemplateComposite/TemplateMethod/src/main/java/ec/edu/espe/templatemethod/model/Tea.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.templatemethod.model;

import java.io.BufferedReader;
import java.io.InputStreamReader;

import ec.edu.espe.templatemethod.model.CaffeineBeverage;

/**
 *
 * @author Carlos Perez,The Javas, DCCO-ESPE
 */
public class Tea extends CaffeineBeverage {

    @Override
    void brew() {
        System.out.println("Steep the tea");
    }

    @Override
    void addCondiments() {
        System.out.println("Adding lemon");
    }

    @Override
    public boolean wantsCondiments() {
        String answer = getUserInput();

        if (answer.toLowerCase().startsWith("y")) {
            return true;
        } else {
            return false;
        }
    }

    private String getUserInput() {
        String answer = "Y";

        System.out.println("Would you like lemon with your tea (y/n)?");
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

        return answer;
    }
}

File: hw/Pérez/U3/HW29TemplateComposite/TemplateMethodJS/model/Tea.js
import { CaffeineBeverage } from './CaffeineBeverage.js';

export class Tea extends CaffeineBeverage {
    brew() {
        console.log("Steeping the tea");
    }

    addCondiments() {
        console.log("Adding lemon");
    }

    wantsCondiments() {
        const answer = this.getUserInput();
        return answer.toLowerCase().startsWith("y");
    }

    getUserInput() {
        const answer = "y"; // Simulación de entrada de usuario
        console.log("Would you like lemon with your tea (y/n)?");
        return answer;
    }
}
